Mathlib/Tactic/Rewrites.lean

```diff
@@ -209,11 +208,12 @@ elab_rules : tactic |
         throwError "Could not find any lemmas which can rewrite the hypothesis {
           ← f.getUserName}"
       for r in results do
-        addRewriteSuggestion tk (← mkConstWithFreshMVarLevels r.name) r.symm
+        addRewriteSuggestion tk [(← mkConstWithFreshMVarLevels r.name, r.symm)]
           r.result.eNew (loc? := .some (.fvar f)) (origSpan? := ← getRef)
       if lucky.isSome then
         match results[0]? with
         | some r => do
+            setMCtx r.mctx
             let replaceResult ← goal.replaceLocalDecl f r.result.eNew r.result.eqProof
             replaceMainGoal (replaceResult.mvarId :: r.result.mvarIds)
```

Comment:
It doesn't really matter here because these are the last things in the `fun`, but in general I think it's nicer to use "dynamic let" versions of setters when they're available since they're easier to reason about:
```suggestion
           withMCtx r.mctx do
              let replaceResult ← goal.replaceLocalDecl f r.result.eNew r.result.eqProof
              replaceMainGoal (replaceResult.mvarId :: r.result.mvarIds)
```